src/monster.cpp

```diff
@@ -1529,7 +1529,9 @@ std::string monster::absorb_hit( const weakpoint_attack &attack, const bodypart_
                                  damage_instance &dam )
 {
     resistances r = resistances( *this );
-    const weakpoint *wp = type->weakpoints.select_weakpoint( attack );
+    weakpoint_attack attack_copy = attack;
+    attack_copy.target = this;
```

Comment:
Is it possible for the target not to be the creature absorbing the hit?

Reply:
No, the target is the creature absorbing the hit.